lumos/optimal_control/convolution.py

```diff
@@ -54,13 +55,54 @@ def __init__(
         self._width = unit_problem.num_in
         self.set_params(params)
 
-    @property
-    def num_con(self):
-        return self._batch * self._unit_problem.num_con
+        # NOTE: here for the constraints, jacobian and hessian, we deliberately leave out
```

Comment:
update this comment: now we ravel everything to ensure the output is flat and consistent

Reply:
done